File: src/frontend/services/api/communication-settings.service.ts
import { apiClient } from '@/lib/axios';

/**
 * Email settings interface
 */
export interface EmailSettings {
  senderName: string;
  replyToEmail: string;
  signature: string;
}

/**
 * SMS settings interface
 */
export interface SmsSettings {
  defaultSender: string;
  optOutMessage: string;
  enableAutoReply: boolean;
}

/**
 * Voice settings interface
 */
export interface VoiceSettings {
  businessHours: string;
  voicemailEnabled: boolean;
  callForwardingNumber?: string;
}

/**
 * Communication settings response interface
 */
export interface CommunicationSettings {
  email?: EmailSettings;
  sms?: SmsSettings;
  voice?: VoiceSettings;
  updatedAt?: string;
}

/**
 * Update communication settings request interface
 * All fields are optional - only provided fields will be updated
 */
export interface UpdateCommunicationSettingsRequest {
  email?: EmailSettings;
  sms?: SmsSettings;
  voice?: VoiceSettings;
}

/**
 * Communication settings service for managing email, SMS, and voice preferences
 */
export const communicationSettingsService = {
  /**
   * Get communication settings for the current business
   */
  get: async (): Promise<CommunicationSettings> => {
    const response = await apiClient.get<CommunicationSettings>('/communication-settings');
    return response.data;
  },

  /**
   * Update communication settings for the current business
   * Only provided fields will be updated (partial update)
   */
  update: async (settings: UpdateCommunicationSettingsRequest): Promise<CommunicationSettings> => {
    const response = await apiClient.put<CommunicationSettings>('/communication-settings', settings);
    return response.data;
  },
};

File: src/backend/QualiFlow.Application/Features/CommunicationSettings/Services/CommunicationSettingsService.cs
using Microsoft.Extensions.Logging;
using QualiFlow.Application.Common.Interfaces;
using QualiFlow.Application.Features.CommunicationSettings.DTOs;
using QualiFlow.Domain.Entities;

namespace QualiFlow.Application.Features.CommunicationSettings.Services;

/// <summary>
/// Service implementation for communication settings operations.
/// </summary>
public partial class CommunicationSettingsService(
    ICommunicationSettingsRepository repository,
    ILogger<CommunicationSettingsService> logger) : ICommunicationSettingsService
{
    /// <inheritdoc />
    public async Task<CommunicationSettingsDto> GetAsync(
        Guid businessId,
        CancellationToken cancellationToken = default)
    {
        LogGettingSettings(logger, businessId);

        var settings = await repository.GetByBusinessIdAsync(businessId, cancellationToken);

        if (settings == null)
        {
            // Return default settings if none exist
            return new CommunicationSettingsDto
            {
                Email = new EmailSettingsDto(),
                Sms = new SmsSettingsDto(),
                Voice = new VoiceSettingsDto(),
            };
        }

        return MapToDto(settings);
    }

    /// <inheritdoc />
    public async Task<CommunicationSettingsDto> UpdateAsync(
        Guid businessId,
        UpdateCommunicationSettingsRequest request,
        CancellationToken cancellationToken = default)
    {
        LogUpdatingSettings(logger, businessId);

        var existing = await repository.GetByBusinessIdAsync(businessId, cancellationToken);

        var settings = existing ?? new Domain.Entities.CommunicationSettings
        {
            BusinessId = businessId,
        };

        // Update email settings
        if (request.Email != null)
        {
            settings.EmailSenderName = request.Email.SenderName;
            settings.EmailReplyTo = request.Email.ReplyToEmail;
            settings.EmailSignature = request.Email.Signature;
        }

        // Update SMS settings
        if (request.Sms != null)
        {
            settings.SmsDefaultSender = request.Sms.DefaultSender;
            settings.SmsOptOutMessage = request.Sms.OptOutMessage;
            settings.SmsEnableAutoReply = request.Sms.EnableAutoReply;
        }

        // Update voice settings
        if (request.Voice != null)
        {
            settings.VoiceBusinessHours = request.Voice.BusinessHours;
            settings.VoiceVoicemailEnabled = request.Voice.VoicemailEnabled;
            settings.VoiceCallForwardingNumber = request.Voice.CallForwardingNumber;
        }

        var saved = await repository.UpsertAsync(settings, cancellationToken);
        return MapToDto(saved);
    }

    private static CommunicationSettingsDto MapToDto(Domain.Entities.CommunicationSettings settings)
    {
        return new CommunicationSettingsDto
        {
            Email = new EmailSettingsDto
            {
                SenderName = settings.EmailSenderName ?? string.Empty,
                ReplyToEmail = settings.EmailReplyTo ?? string.Empty,
                Signature = settings.EmailSignature ?? string.Empty,
            },
            Sms = new SmsSettingsDto
            {
                DefaultSender = settings.SmsDefaultSender ?? string.Empty,
                OptOutMessage = settings.SmsOptOutMessage ?? "Reply STOP to unsubscribe",
                EnableAutoReply = settings.SmsEnableAutoReply,
            },
            Voice = new VoiceSettingsDto
            {
                BusinessHours = settings.VoiceBusinessHours ?? string.Empty,
                VoicemailEnabled = settings.VoiceVoicemailEnabled,
                CallForwardingNumber = settings.VoiceCallForwardingNumber,
            },
            UpdatedAt = settings.UpdatedAt,
        };
    }

    // ============================================================================
    // Logger Message Delegates
    // ============================================================================

    [LoggerMessage(
        EventId = 1,
        Level = LogLevel.Information,
        Message = "Getting communication settings for business {BusinessId}")]
    private static partial void LogGettingSettings(ILogger logger, Guid businessId);

    [LoggerMessage(
        EventId = 2,
        Level = LogLevel.Information,
        Message = "Updating communication settings for business {BusinessId}")]
    private static partial void LogUpdatingSettings(ILogger logger, Guid businessId);
}

